pandas/core/arrays/datetimelike.py

```diff
@@ -79,19 +85,270 @@ def _get_attributes_dict(self):
         """return an attributes dict for my class"""
         return {k: getattr(self, k, None) for k in self._attributes}
 
+    @property
+    def _scalar_type(self):
+        """The scalar associated with this datelike
+
+        * PeriodArray : Period
+        * DatetimeArray : Timestamp
+        * TimedeltaArray : Timedelta
+        """
+        # type: # () -> Union[type, Tuple[type]]
+        raise AbstractMethodError(self)
+
+    def _scalar_from_string(self, value):
+        # type: (str) -> Union[Period, Timestamp, Timedelta]
+        raise AbstractMethodError(self)
+
+    def _unbox_scalar(self, value):
+        """
+        Unbox the integer value of a scalar `value`.
+
+        Parameters
+        ----------
+        value : Union[Period, Timestamp, Timedelta]
+
+        Returns
+        -------
+        int
+
+        Examples
+        --------
+        >>> self._unbox_scalar(Timedelta('10s'))  # DOCTEST: +SKIP
+        10000000000
+        """
+        # TODO: handle NAT?
+        raise AbstractMethodError(self)
 
-class DatetimeLikeArrayMixin(ExtensionOpsMixin, AttributesMixin):
+    def _check_compatible_with(self, other):
+        """
+        Verify that `self` and `other` are compatible.
+
+        Used in
+
+        * __setitem__
+
+        Parameters
+        ----------
+        other
+
+        Raises
+        ------
+        Exception
+        """
+        raise AbstractMethodError(self)
+
+
+class DatelikeOps(ExtensionOpsMixin):
```

Comment:
This and TimelikeOps seem like they could be broken off in a standalone PR.  Would taking that off your plate make things easier?

Reply:
I tried this earlier, but apparently gave up too soon. I just hacked something together and was able to import pandas. The only thing giving me pause is PeriodIndex/PeriodArray. That may require some special care until this PR is in.

I'll clean things up and split this section out into a new one.

Reply:
This is done in ##24038